### PR TITLE
Migrate sample-README rendering from marked to Sätteri (markdownToMdast)

### PR DESCRIPTION
**Pull request by IEvangelist**
## What

Follow-up to the Astro 7 upgrade (#1309). Migrates the **sample-README renderer** from the standalone [`marked`](https://github.com/markedjs/marked) parser to **Sätteri**, the Rust-based Markdown engine Astro 7 ships as default.

- Swaps `marked.lexer(...)` for Sätteri's `markdownToMdast(...)` entry point and walks standard **MDAST** nodes instead of marked tokens.
- `markdownToMdast` is a standalone parser function from the `satteri` package, so this change is **independent of the Astro `markdown.processor`** — no config-pipeline change required.
- Component composition is preserved 1:1: `<Code>`, `<Steps>`, `<Aside>`, mermaid, ordered/unordered lists, and GitHub alerts all render identically. Inline/leftover content is converted via `mdast-util-to-hast` + `hast-util-to-html` with a custom link handler (relative-href resolution + external `target="_blank"`).
- Heading sentence-casing, slug/anchor generation, and standalone-image gallery extraction are ported to MDAST node fields (`url`/`alt`/`value`/`children`).

Also removes the **orphaned `starlight-llms-txt@0.8.1.patch`** — the `customSelectors` twoslash-stripping feature was upstreamed and ships natively since `starlight-llms-txt@0.10.0` (bumped in #1309). The patch file was left on disk but already de-referenced from `pnpm-workspace.yaml`/lockfile. Stale "patch" wording in the llms-txt sentinel tests was reworded to reflect the native feature; **all assertion values and logic are unchanged**.

## Deferred (not in this PR)

The **core `unified()` → `satteri()` processor swap** is **blocked** by `starlight-image-zoom@0.15.0`, which hard-throws at `astro:config:setup` under `processor: satteri()`:

> The Sätteri Markdown processor is not yet supported by the starlight-image-zoom plugin. Switch to 'unified()'.

Sätteri support is unreleased upstream — tracked in [HiDeoo/starlight-image-zoom#63](https://github.com/HiDeoo/starlight-image-zoom/issues/63) (waiting on a satteri dependency). image-zoom powers image zoom across the docs and the sample gallery, so it can't simply be dropped. Verified it is the **only** config-level blocker (`astro sync` passes once image-zoom is disabled).

`astro.config.mjs` is therefore **unchanged** in this PR — the site keeps rendering exactly as today with **zero functionality regression**. The core swap will be redone as a follow-up once image-zoom ships Sätteri support.

## Dependency changes

| Action | Package |
| --- | --- |
| Removed | `marked` |
| Added (deps) | `satteri`, `mdast-util-to-hast`, `hast-util-to-html` |
| Added (devDeps) | `@types/mdast` |
| Untouched | `@astrojs/markdown-remark`, `remark-directive` (still used by the `unified()` pipeline) |

## Validation

- Full unit suite green: **22 files, 244 tests passed** (components, llms-txt, llms-small, twoslash-blocks, api-markdown, sample-readme-headings, …).
- `eslint .` clean; `astro sync` passes.
- `pnpm build` intentionally left to CI per repo policy.

## Notes for reviewers

The substantive change is `src/frontend/src/components/SampleReadmeBlocks.astro` (the recursive MDAST → Astro renderer) — heading HTML is kept byte-identical and asserted by `tests/unit/custom-components.vitest.test.ts`.

### CODE REVIEW

File: src/frontend/src/components/SampleReadmeBlocks.astro [hunk 1]
```diff
@@ -53,77 +65,86 @@ function resolveReadmeHref(href: string): string {
...
+/**
+ * Custom `mdast-util-to-hast` handler for links. Mirrors the previous marked
+ * `renderer.link`: README links are resolved against the sample's source path
+ * (so `./relative` paths point at the right file on GitHub) and external links
+ * open in a new tab with a safe `rel`.
+ */
+function linkHandler(state: State, node: Link): HastElement {
+  const resolvedHref = resolveReadmeHref(node.url ?? '');
+  const isExternal = /^(?:https?:)?\/\//i.test(resolvedHref);
+  const properties: Record<string, string> = { href: resolvedHref };
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply src/frontend/src/components/SampleReadmeBlocks.astro]: Good catch — fixed in 0f4b8848. `linkHandler` now guards against a falsy/whitespace-only `node.url` and returns `undefined`, so links with no destination (e.g. `[]()`) are dropped entirely rather than rewritten to a broken `${sampleHref}/` URL. This restores the previous marked renderer's behavior, which returned an empty string for a falsy `href`. Also added a regression assertion to the SampleDetail render test covering `[broken empty link]()`.

**Review by davidfowl — APPROVED** [review]
